sdks/keyvault/src/client.rs

```diff
@@ -15,109 +12,43 @@ pub(crate) const API_VERSION: &str = "7.0";
 ///
 /// ```no_run
 /// use azure_keyvault::KeyVaultClient;
-/// let client = KeyVaultClient::new(&"{client_id}", &"{client_secret}", &"{tenant_id}", &"test-keyvault");
+/// use azure_auth_aad::DefaultCredential;
+/// let creds = DefaultCredential::default();
+/// let client = KeyVaultClient::new(&creds, &"test-keyvault");
 /// ```
 #[derive(Debug)]
-pub struct KeyVaultClient<'a> {
-    pub(crate) aad_client_id: &'a str,
-    pub(crate) aad_client_secret: &'a str,
-    pub(crate) aad_tenant_id: &'a str,
+pub struct KeyVaultClient<'a, T> {
+    pub(crate) token_credential: &'a T,
     pub(crate) keyvault_name: &'a str,
```

Comment:
Why are we borrowing the keyvalue_name and token_credential instead of just taking ownership?

Reply:
It was the original implementation and I left it as it was.

Reply:
It would be ergonomic to leave the choice to the caller, for example using `Cow`. For example CosmosDB uses this pattern:

[https://github.com/Azure/azure-sdk-for-rust/blob/8e5f1330c6f0c5334678f39bd7358748afc306ff/sdks/cosmos/src/clients/collection_struct.rs#L24-L37](https://github.com/Azure/azure-sdk-for-rust/blob/8e5f1330c6f0c5334678f39bd7358748afc306ff/sdks/cosmos/src/clients/collection_struct.rs#L24-L37)

The only limitation is the *cloneability* of the inner type, something like:

```rust
pub struct KeyVaultClient<'a, T: Clone> {
    pub(crate) token_credential: Cow<'a, T>,
```

Or we can try the [MaybeOwned](https://docs.rs/maybe-owned/0.3.4/maybe_owned/enum.MaybeOwned.html) crate that relaxes this limitation. We should probably leave this as a separate PR.

Reply:
I'm not sure we really need to care too deeply about the performance implications here. Taking a String and risky some additional heap allocations is fine IMO